File: src/main/java/myratpackexamples/webpoll/PollApplication.java
package myratpackexamples.webpoll;

import ratpack.guice.Guice;
import ratpack.handling.Chain;
import ratpack.server.RatpackServer;
import ratpack.server.RatpackServerSpec;

public class PollApplication {
    private static Chain addHandlersToChain(Chain chain) {
        return chain
                .post("poll", CreatePollHandler.class)
                .get("poll/:poll", RetrievePollHandler.class);
    }

    public static RatpackServerSpec setupServer(RatpackServerSpec server) {
        return server
                .registry(Guice.registry(b -> b.module(PollModule.class)))
                .handlers(PollApplication::addHandlersToChain);
    }

    public static void main(String... args) throws Exception {
        new InMemoryMongoDb();
        RatpackServer.start(PollApplication::setupServer);
    }
}

File: src/main/java/myratpackexamples/webpoll/PollApplication.kt
package myratpackexamples.webpoll

import myratpackexamples.webpoll.createpoll.CreatePollHandler
import myratpackexamples.webpoll.createvote.CreateVoteHandler
import ratpack.guice.Guice
import ratpack.handling.Chain
import ratpack.server.RatpackServer
import ratpack.server.RatpackServerSpec

private fun addHandlersToChain(chain: Chain): Chain = chain
        .post("poll", CreatePollHandler::class.java)
        .get("poll/:poll", RetrievePollHandler::class.java)
        .post("poll/:poll/vote", CreateVoteHandler::class.java)

fun setupServer(server: RatpackServerSpec): RatpackServerSpec = server
        .registry(Guice.registry { it.module(PollModule::class.java) })
        .handlers { addHandlersToChain(it) }

fun main(args: Array<String>) {
    InMemoryMongoDb()
    RatpackServer.start { setupServer(it) }
}